Common/src/geometry_structure.cpp

```diff
@@ -14743,6 +14766,27 @@ void CPhysicalGeometry::SetPeriodicBoundary(CConfig *config) {
   
 }
 
+void CPhysicalGeometry::SetCustomBoundary(CConfig *config){
```

Comment:
It is a little peculiar for me to see physical BCs set in the geometry class.. Is there a way to put these into the solver class(es), or is there currently an issue with data encapsulation or access that disallows this?

Reply:
I put it there for easy access to "node" and "vertex" arrays. However I don't see any issue if I put it into the CSolver and CEulerSolver classes. I just have to pass an additional CGeometry as argument in order to keep the access to node and vertex.

Note that this routine is for now called only at SU2 initialization, just to guarantee that if the user do not provide BC values from Python, the computation can still run with the default values read from the config file.

Reply:
I see.. yes, you could pass in the geometry object to achieve this.

My main concerns are 1) adding extra doubles to the CPoint class (remember that every node in the grid then has more data, when we only need this information on the boundaries) and 2) having PDE-related information in the CGeometry classes when it is more natural in the CSolver classes.

This is a really nice feature, and I have been thinking more about it. Is it possible to generalize it by putting the "CustomBC" routines up in the CSolver parent class, so that all of the child solver classes can inherit the ability to have custom BCs from Python? You would have to allocate some memory to store the values, ideally just for the boundary vertices, like the data structures at the end of the mean flow solver constructor. The data could be stored in generic "custom" structures and then the child classes could access them in the BC routines when necessary for solver-specific tasks (for CHT, custom profiles, etc.). That way, you really only need to implement it one time. What do you think?